mobile/src/main/java/com/gelakinetic/mtgfam/fragments/ImportFragment.java

```diff
@@ -0,0 +1,273 @@
+            SnackbarWrapper.makeAndShowText(getFamiliarActivity(), R.string.empty_deck_text, SnackbarWrapper.LENGTH_LONG);
+            mImportStarted = false;
+            return;
+        }
+
+        /* TODO: check if deck already exists */
+
+        activity.setLoading();
+
+        final String name = String.valueOf(getDeckNameInput());
+        final String lines = String.valueOf(getDeckTextInput());
+
+        mImportTask = new ImportTask(name, lines).execute();
+    }
+
+    @Override
+    public void onDestroy() {
+
+        super.onDestroy();
+
+        if (mImportTask != null) {
+            mImportTask.cancel(true);
+            mImportTask = null;
+        }
+    }
+
+    /**
+     * @param menu     The options menu in which you place your items.
+     * @param inflater The inflater to use to inflate the menu
+     */
+    @Override
+    public void onCreateOptionsMenu(@NotNull Menu menu, @NotNull MenuInflater inflater) {
+
+        super.onCreateOptionsMenu(menu, inflater);
+    }
+
+    /* TODO: make class static to prevent leaks? */
+    private class ImportTask extends AsyncTask<Void, String[], DeckListImporter> {
+
+        private final String mLines;
+        private final String mName;
+        private final ArrayList<MtgCard> unknownCards = new ArrayList<>();
+        private final ArrayList<MtgCard> importedCards = new ArrayList<>();
+
+        ImportTask(String name, String lines) {
+            mName = name;
+            mLines = lines;
+        }
+
+        @Override
+        protected DeckListImporter doInBackground(Void... voids) {
+            final DeckListImporter importer = new DeckListImporter();
+            try (BufferedReader br = new BufferedReader(new StringReader(mLines))) {
+                String line;
+                while ((line = br.readLine()) != null) {
+                    importer.parseLine(line);
+                }
+            } catch (IOException e) {
+                /* TODO: show some kind of message here?
+                 * I don't think StringReader actually throws unless it's closed... */
+                return importer;
+            }
+            /* store local copy of cards so they can be updated */
+            List<MtgCard> cardList = importer.getParsedCards();
+            /* match cards with database */
+            try {
+                /* TODO: does error snackbar get shown properly? We're not on UI thread... */
+                //Issues happen here if ImportTask is made static
+                MtgCard.initCardListFromDb(getContext(), cardList);
+            } catch (FamiliarDbException fde) {
+                handleFamiliarDbException(false);
+            }
+
+            /* find out which cards are known */
+            for (MtgCard card : cardList) {
+                if (card.getMultiverseId() == 0) {
+                    try {
+                        MtgCard noSetCard = new MtgCard(getActivity(), card.getName(), null, card.mIsFoil, card.mNumberOf);
```

Comment:
When `noSetCard` is created from `card`, it's not copying the `mIsSideboard` property, so it's importing too many cards into the main deck and too few in the sideboard.